snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshot.java

```diff
@@ -97,32 +113,59 @@ public String getId() {
     return metadata.getSnapshotIdAsString();
   }
 
+  @Override
+  public long getChecksum() {
+    return checksum;
+  }
+
   @Override
   public void close() {
     // nothing to be done
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(getDirectory(), getMetadata());
+    int result = getDirectory().hashCode();
```

Comment:
Why do you have to calculate hash like this?

Reply:
I auto generated the hash using the IntelliJ defaults. I thought we'd discussed this in a team meeting and agreed to just use that in the future? I didn't manually write it.

Reply:
:+1: Never seen this autogenerated code. Is it the default in IntelliJ?

Reply:
When you generate it, it asks you how you want to generate it, and one of the options is called `IntelliJ default`. This is what it generates for me :man_shrugging: